### PR TITLE
Implement fading tracker.

### PR DESCRIPTION
**Pull request by sjonany**
Tracker fades over time so user can continuous align expectation on how fast to read. More details in https://app.clickup.com/t/9cd85x.
Btw, you can stop the fading animation partially too.
Box shadow doesn't fade out in the current implementation, and... maybe that's a good thing.

### CODE REVIEW

File: extension/content.css [hunk 1]
```diff
@@ -1,16 +1,15 @@
-mark {
-}
 .markedYellow {
-	background-color:yellow!important;
+	background-color: yellow;
```
Comment (sjonany) [inline extension/content.css:2]: Need to remove !important, or else the fadeaway style will always get overwritten.

File: extension/content.css [hunk 2]
```diff
@@ -1,16 +1,15 @@
...
 }
 .markedBlue {
-	background-color:#c9daf8 !important; /* Same shade of blue as when normally highlight in browser */
+	/* Same shade of blue as when normally highlight in browser */
+	background-color: rgb(201, 218, 248);
 }
 .markedLightBlue {	
-	background-color:#e6effd !important; /* Lighter blue */
+	background-color: rgb(230, 239, 253); /* Lighter blue */
```
Comment (sjonany) [inline extension/content.css]: Changed to rgb syntax cos it helped me debug previously.
If you have a strong pref for the # syntax lmk and ill revert.

File: extension/content_script.js [hunk 3]
```diff
@@ -64,7 +63,9 @@ If a tracker is currently moving, stop it.
 See startMove()
 */
 function stopMove() {
-	if (timer) { 
+	if (timer) {
+		// Stop fading animation.
+		$("mark").stop();
```
Comment (sjonany) [inline extension/content_script.js:72]: This is what allows partial fade.
When u lift key up, whatever fading animation is stopped there, indicating partial progress.

Comment (sjonany) [reply extension/content_script.js:72]: Not true partial progress tho, cos when user resumes, we immediately go to next sentence

File: extension/content_script.js [hunk 4]
```diff
@@ -18,7 +18,6 @@ var tracker = null;
 var timer = null;
 var speed = 10; // Base speed, not accounting for sentence length; adjustable w/ D/S
 var speed_bias = 500; // Minimum amount of speed spent on each sentence (in milliseconds)
-var speed_adj = 0; // Speed after it has been adjusted by sentence length
```
Comment (sjonany) [inline extension/content_script.js:25]: don't need this. See calculateTrackerLife().
I'm trying to minimize the number of global variables and the bookkeeping cost of having to maintain and making sure they're consistent.

File: extension/content_script.js [hunk 5]
```diff
@@ -97,15 +98,21 @@ function startMove(dir) { // Note: I have combined the "moveUp" and "moveDown" f
...
+// Calculate lingering time for current tracker in ms.
+function calculateTrackerLife() {
+	return (speed * tracker.getTrackerLen()) + speed_bias;
+}
+
 // Move one sentence up
 function moveUpOne() { // Sets start and end
 	tracker.movePrevious();
 	highlight(tracker);
-	speed_adj = (speed * tracker.getTrackerLen()) + speed_bias;
```
Comment (sjonany) [inline extension/content_script.js:112]: now that it's a getter function, less worry about recalculating this on every code path that changes speed.

File: extension/content.css (hunk 2 above)
Comment (benfrancis314) [reply extension/content.css]: Nope I actually prefer rgb as well! Easier to move between rgba for experimenting with opacity too

Comment (benfrancis314) [reply extension/content.css:9]: LGTM

File: extension/content.css (hunk 1 above)
Comment (benfrancis314) [reply extension/content.css:2]: LGTM

File: extension/content_script.js (hunk 4 above)
Comment (benfrancis314) [reply extension/content_script.js:25]: Yep v good idea and thanks for identifying this as good candidate for eliminating global variable

File: extension/content_script.js (hunk 5 above)
Comment (benfrancis314) [reply extension/content_script.js:108]: LGTM

File: extension/content_script.js (hunk 3 above)
Comment (benfrancis314) [reply extension/content_script.js:72]: Very cool! Definitely helps jarring nature of auto-read mode. Next step is probably getting "fade in" and also integrating for manual-read mode

File: extension/content.css (hunk 1 above)
Comment (benfrancis314) [reply extension/content.css:2]: LGTM

File: extension/content_script.js [hunk 6]
```diff
@@ -154,10 +161,25 @@ function highlight(tracker) {
     	start: tracker.getStart(),
     	length: tracker.getEnd() - tracker.getStart()
 	}], {
-		className: 'marked'
+		className: currentStyle
 	});
 };
 
+/*
```
Comment (benfrancis314) [inline extension/content_script.js:168]: Looks great!

File: extension/content_script.js [hunk 7]
```diff
@@ -154,10 +161,25 @@ function highlight(tracker) {
...
+Fade the current tracker indicator according to the calculated speed.
+*/
+function fadeTracker() {
```
Comment (benfrancis314) [inline extension/content_script.js:171]: Looks great!

**Review by benfrancis314 — APPROVED** [review]
LGTM

File: extension/content_script.js (hunk 3 above)
Comment (sjonany) [reply extension/content_script.js:72]: Tracked in clickup - experiment w/ transition animation.

**Pull request merged**